### PR TITLE
FIX: Hovering reaction icon causing error would flood AJAX requests

### PR DESCRIPTION
**Pull request by martin-brennan**
If a user hovered over a reaction to see a list of users,
mostly a problem when anon browsing (see
https://github.com/discourse/discourse/commit/5935148bd863ea80865cee1049b3a44f3fbf3cc8),
we didn't have a throttle when erroring, so extra hovers would cause
error messages.

This fixes the issue by adding a throttle, and also adds some general
reactions system testing for this area.

### CODE REVIEW

File: assets/javascripts/discourse/widgets/discourse-reactions-list-emoji.js [hunk 1]
```diff
@@ -109,4 +123,10 @@ export default createWidget("discourse-reactions-list-emoji", {
       }
     });
   },
+
+  _allowHover() {
```
Comment (martin-brennan) [inline assets/javascripts/discourse/widgets/discourse-reactions-list-emoji.js:120]: Yes this is not ideal, but there is seemingly no way to get rid of `(hover: none)` in chrome headless. Maybe in time we could globally set `window.isSystemTest` in system tests, like how we have `isTesting()` available from ember

File: assets/javascripts/discourse/widgets/discourse-reactions-list-emoji.js [hunk 2]
```diff
@@ -16,11 +16,25 @@ export default createWidget("discourse-reactions-list-emoji", {
...
-        this.callWidgetFunction("getUsers", this.attrs.reaction.id);
+      if (
+        !this.attrs.users?.length &&
+        !this.debounceLoadReactions &&
+        !this.loadingReactions
+      ) {
+        this.loadingReactions = true;
+        this.callWidgetFunction("getUsers", this.attrs.reaction.id)
+          .catch(() => {
+            this.debounceLoadReactions = true;
```
Comment (Drenmi) [inline assets/javascripts/discourse/widgets/discourse-reactions-list-emoji.js]: Any chance we can use [`#debounce`](https://api.emberjs.com/ember/release/functions/@ember%2Frunloop/debounce) from `@ember/runloop`? 🤔

File: assets/javascripts/discourse/widgets/discourse-reactions-list-emoji.js (hunk 1 above)
Comment (Drenmi) [reply assets/javascripts/discourse/widgets/discourse-reactions-list-emoji.js:120]: > Maybe in time we could globally set `window.isSystemTest` [...]

👍

**Review by Drenmi — APPROVED** [review]
LGTM! 🚀

File: assets/javascripts/discourse/widgets/discourse-reactions-list-emoji.js (hunk 2 above)
Comment (martin-brennan) [reply assets/javascripts/discourse/widgets/discourse-reactions-list-emoji.js]: Yes this works! Using that instead :)

**Pull request merged**